pkg/controller/cluster/http_handlers.go

```diff
@@ -116,3 +128,210 @@ func validateBucketName(bucket string) (bool, error) {
+		return
+	}
+
+	// Only one tenant is allowed in a single namespace, gathering the first tenant in the list
+	tenant := tenants.Items[0]
+
+	tenantConfiguration, err := c.getTenantCredentials(ctx, &tenant)
+	if err != nil {
+		if errors.Is(err, ErrEmptyRootCredentials) {
+			writeSTSErrorResponse(ctx, w, true, ErrSTSInternalError, fmt.Errorf("Tenant '%s' is missing root credentials", tenant.Name))
+			return
+		}
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInternalError, fmt.Errorf("Error getting tenant '%s' root credentials: %s", tenant.Name, err))
+		return
+	}
+	adminClient, err := tenant.NewMinIOAdmin(tenantConfiguration, c.getTransport())
+	if err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSInternalError, fmt.Errorf("Error communicating with tenant '%s': %s", tenant.Name, err))
+		return
+	}
+
+	// Session Policy
+	sessionPolicyStr := r.Form.Get(stsPolicy)
+	var compactedSessionPolicy string
+	var sessionPolicy *iampolicy.Policy
+	if len(sessionPolicyStr) > 0 {
+		compactedSessionPolicy, err = miniov2.CompactJSONString(sessionPolicyStr)
+		if err != nil {
+			writeSTSErrorResponse(ctx, w, true, ErrSTSMalformedPolicyDocument, err)
+			return
+		}
+		sessionPolicy, err = iampolicy.ParseConfig(bytes.NewReader([]byte(compactedSessionPolicy)))
+		if err != nil {
+			writeSTSErrorResponse(ctx, w, true, ErrSTSMalformedPolicyDocument, err)
+			return
+		}
+		// Version in policy must not be empty
+		if sessionPolicy.Version == "" {
+			writeSTSErrorResponse(ctx, w, true, ErrSTSInvalidParameterValue, fmt.Errorf("Invalid session policy version"))
+			return
+		}
+		// The plain text that you use for both inline and managed session
+		// policies shouldn't exceed 2048 characters.
+		if len(compactedSessionPolicy) > 2048 {
+			writeSTSErrorResponse(ctx, w, true, ErrSTSPackedPolicyTooLarge, fmt.Errorf("Session policy should not exceed 2048 characters"))
+			return
+		}
+	}
+
+	var bfPolicy iampolicy.Policy
+	for _, pb := range policyBindings {
+		if sessionPolicy != nil {
+			bfPolicy = bfPolicy.Merge(*sessionPolicy)
+		}
+		for _, policyName := range pb.Spec.Policies {
+			policy, err := GetPolicy(ctx, adminClient, policyName)
+			if err != nil {
+				klog.Error(fmt.Errorf("Invalid policy %s, ignoring: %s", policyName, err))
+				continue
+			}
+			parsedPolicy, err := iampolicy.ParseConfig(bytes.NewReader([]byte(policy.Policy)))
+			if err != nil {
+				klog.Error(fmt.Errorf("Invalid policy, not parseable %s, ignoring: %s", policyName, err))
+				continue
+			}
+			bfPolicy = bfPolicy.Merge(*parsedPolicy)
+		}
+	}
+	bfJSONPolicy, _ := json.Marshal(bfPolicy)
+	bfCompact, err := miniov2.CompactJSONString(string(bfJSONPolicy))
+	if err != nil {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSMalformedPolicyDocument, err)
+		return
+	}
+	if len(bfCompact) > 2048 {
+		writeSTSErrorResponse(ctx, w, true, ErrSTSPackedPolicyTooLarge, fmt.Errorf("PolicyBinding resulting policy is too long, Policy should not exceed 2048 characters"))
```

Comment:
perhaps add to the error the current length of the combined policy

Reply:
added